addons/account/models/account.py

```diff
@@ -539,19 +539,37 @@ class AccountGroup(models.Model):
     _name = "account.group"
     _description = 'Account Group'
     _parent_store = True
-    _order = 'code_prefix'
+    _order = 'code_prefix_start'
 
     parent_id = fields.Many2one('account.group', index=True, ondelete='cascade')
     parent_path = fields.Char(index=True)
     name = fields.Char(required=True)
-    code_prefix = fields.Char()
+    code_prefix_start = fields.Char()
+    code_prefix_end = fields.Char()
+    company_id = fields.Many2one('res.company', required=True, default=lambda self: self.env.company)
+
+    _sql_constraints = [
+        (
+            'check_length_prefix',
+            'CHECK(char_length(COALESCE(code_prefix_start, \'\')) = char_length(COALESCE(code_prefix_end, \'\')))',
+            'The length of the starting and the ending code prefix must be the same'
+        ),
+    ]
+
+    @api.onchange('code_prefix_start', 'code_prefix_end')
+    def _onchange_code_prefix(self):
+        if self.code_prefix_start and (not self.code_prefix_end or self.code_prefix_end < self.code_prefix_start):
+            self.code_prefix_end = self.code_prefix_start
+        if self.code_prefix_end and (not self.code_prefix_start or self.code_prefix_start > self.code_prefix_end):
+            self.code_prefix_start = self.code_prefix_end
 
     def name_get(self):
         result = []
         for group in self:
-            name = group.name
-            if group.code_prefix:
-                name = group.code_prefix + ' ' + name
+            prefix = group.code_prefix_start
+            if prefix and group.code_prefix_end != group.code_prefix_start:
+                prefix += '-' + group.code_prefix_end
```

Comment:
not sure this is something generic

Reply:
What do you mean? The hyphen between the two prefixes?